x-pack/plugins/code/server/plugin.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { first } from 'rxjs/operators';
+import { TypeOf } from '@kbn/config-schema';
+import { LoggerFactory, PluginInitializerContext, RecursiveReadonly } from 'src/core/server';
+import { deepFreeze } from '../../../../src/core/utils';
+import { CodeConfigSchema } from './config';
+
+/**
+ * Describes public Code plugin contract returned at the `setup` stage.
+ */
+export interface PluginSetupContract {
+  legacy: {
```

Comment:
Optional: Could you add  `/** @deprecated */` annotation ? That shows it's not intended to be used as a public API.

Reply:
done